SPECS/kernel/perf_bpf_test_add_nonnull_argument.patch

```diff
@@ -0,0 +1,35 @@
+From 81807bcba96f3d98824e9fa2c597c22eb619dd7a Mon Sep 17 00:00:00 2001
+From: Rachel Menge <rachelmenge@microsoft.com>
+Date: Fri, 17 Nov 2023 05:21:58 +0000
+Subject: [PATCH] Fix perf bpf tests
+
+Taken from: 
+https://lore.kernel.org/all/ZMj8+bvN86D0ZKiB%40kernel.org/
```

Comment:
It's generally not a good idea to pull patches from mailing list archives; that skips the very, very important steps of peer review and upstream acceptance for the patch. In other words, just because a patch is in a mailing list archive doesn't mean it's upstream.

Also, the best way to generate a patch file from an upstream commit (for any project, not just the kernel) is to create the patch file with git format-patch; since this particular commit upstream is ed847e30f001b207013b6136c264454d7560557f, the right way to do this is `git format-patch -1 ed847e30f001b207013b6136c264454d7560557f`. That brings along all the commit description as well as retaining the patch authorship and, most importantly, it retains the actual upstream commit id (on the first line, e.g. `From ed847e30f001b207013b6136c264454d7560557f Mon Sep 17 00:00:00 2001`)

Reply:
Makes sense! Updated patch to contain upstream info and will look to use git format-patch -1 in future :)